File: info/1. Алгоритмы/2. Сортировка/10. Сортировка подсчетом/code sample/sample.py
def counting_sort(sequince):
    min_value = min(sequince)
    max_value = max(sequince)
    support = [0 for i in range(max_value-min_value+1)]
    for element in sequince:
        support[element-min_value] += 1
    index = 0
    for i in range(len(support)):
        for element in range(support[i]):
            sequince[index] = i+min_value
            index += 1
    return None


sequince = [5, 0, -2, 7, 3, -2]
print(sequince)

counting_sort(sequince)
print(sequince)

File: info/1. Алгоритмы/2. Сортировка/10. Сортировка подсчетом/code sample/Sample.java
package sample1;

import java.util.Arrays;

public class Sample {

    public static void main(String[] args) {

        int[] sort = new int[] { 5, 0, -2, 7, 3 };
        System.out.println(Arrays.toString(sort));
        countingSort(sort);
        System.out.println(Arrays.toString(sort));

    }

    public static void countingSort(int[] sort) {
        int[] minMax = findMinMax(sort);
        int minValue = minMax[0];
        int maxValue = minMax[1];
        int[] support = new int[maxValue - minValue + 1];
        for (int element : sort) {
            support[element - minValue] += 1;
        }
        int index = 0;
        for (int i = 0; i < support.length; i++) {
            for (int j = 0; j < support[i]; j++) {
                sort[index] = i + minValue;
                index += 1;
            }

        }
    }

    public static int[] findMinMax(int[] array) {
        int min = array[0];
        int max = array[0];
        for (int element : array) {
            if (min > element) {
                min = element;
            }
            if (max < element) {
                max = element;
            }
        }
        return new int[] { min, max };
    }
}
